### PR TITLE
[5.3] Add ability to pass an array to Notification Fakes

### PR DESCRIPTION
**Pull request by spyric**
In documentation in [Notification Fakes](https://laravel.com/docs/5.3/mocking#notification-fakes) section wrote:

```
// Assert a notification was sent to the given users...
Notification::assertSentTo(
    [$user], OrderShipped::class
);
```

But when passing array to first argument it throws an Exception:

```
BadMethodCallException: Method getKey does not exist.
```

This PR adds support to pass an array as argument

If you would like to merge it please add #hacktoberfest tag

### CODE REVIEW

**Comment by taylorotwell** [comment]
Just change the array check. Don't change to a flatMap or change any other logic. It pollutes the PR and makes it hard to tell what is actually being changed.

**Comment by spyric** [comment]
@taylorotwell I have fixed some logic now instead of `flatMap`, `map` is using - I'm getting array with notifications for notiffiable. And that check:
In notification should not be sent - There should be no non-empty arrays
In notification should be sent - There should be no empty arrays

**Comment by taylorotwell** [comment]
I don't understand why there are so many changes other than just converting an array to a Collection?

**Comment by spyric** [comment]
Before this PR function works with one notifiable and return notifications array that suits to type and closure.

If we are passing an array of notifiables and apply previous algorithm for each item we will get an array of notifications array.

File: src/Illuminate/Support/Testing/Fakes/NotificationFake.php [hunk 1]
```diff
@@ -58,23 +66,29 @@ public function assertNotSentTo($notifiable, $notification, $callback = null)
...
+
+            return $notifications->filter(function ($arguments) use ($callback) {
+                return $callback(...array_values($arguments));
+            })->pluck('notification');
+        });
     }
 
     /**
-     * Determine if there are more notifications left to inspect.
+     * Determine if there are morae notifications left to inspect.
```
Comment (taylorotwell) [inline src/Illuminate/Support/Testing/Fakes/NotificationFake.php]: Why?

**Comment by taylorotwell** [comment]
I've fixed it in a different way that doesn't require changes to the sent method.

**Pull request closed without merging**